geonetwork/gn-tool/get-uuids.py

```diff
@@ -1,10 +1,20 @@
 from xml.etree.ElementTree import fromstring, ElementTree
 import sys
+from requests.auth import HTTPBasicAuth
 import unicodedata
 
 import requests
 
-text_xml = requests.get(sys.argv[1]).text
+if len(sys.argv) > 2:
+    auth = HTTPBasicAuth(sys.argv[2], sys.argv[3])
+    if len(sys.argv) >= 5:
+        headers = {"X-XSRF-TOKEN": sys.argv[4]}
+    else:
+        headers = None
+else:
+    auth = None
+
```

Comment:
`
auth = None
headers = None
if len(sys.argv) > 2:
    auth = HTTPBasicAuth(sys.argv[2], sys.argv[3])
    if len(sys.argv) >= 5:
        headers = {"X-XSRF-TOKEN": sys.argv[4]}
`
I tried to run it without a username/password and it broke because in that case the `headers` variable doesn't get set at all and fails the requests call later on.

Reply:
Since argparse is in the standard library, there's basically no reason to access sys.argv directly, it makes argument parsing way more reliable.

https://docs.python.org/2.7/library/argparse.html#example

Example here (not tested): https://github.com/aodn/utilities/pull/181